spec/currency_spec.rb

```diff
@@ -186,6 +187,27 @@ def to_s
       it "raises UnknownCurrency with unknown currency" do
         expect { Currency.new("xxx") }.to raise_error(Currency::UnknownCurrency, /xxx/)
       end
+
+      it 'returns old object for the same :key' do
+        expect(Currency.new("USD").object_id).to be(Currency.new("USD").object_id)
+        expect(Currency.new("USD").object_id).to be(Currency.new(:usd).object_id)
+        expect(Currency.new("USD").object_id).to be(Currency.new(:USD).object_id)
+        expect(Currency.new("USD").object_id).to be(Currency.new('usd').object_id)
+        expect(Currency.new("USD").object_id).to be(Currency.new('Usd').object_id)
```

Comment:
oh, i mean `expect(Currency.new("USD")).to be(Currency.new("USD"))`. `be` checks that objects are the same.